### ISSUE TITLE
Support for concat of files with unequal length in concat dim

### ISSUE THREAD

**Issue by cisaacstern** [intake/fsspec-reference-maker#76]
@jbusecke and I are wondering if 
https://github.com/intake/fsspec-reference-maker/blob/89d32322fdd3ba631a4fe846475a8857f10ad872/fsspec_reference_maker/combine.py#L313-L316

should instead check that `v / ds0.dims[k] > 1`.

We were working with the Pangeo Forge [`HDFReference`](https://github.com/pangeo-forge/pangeo-forge-recipes/blob/c4a4ed84027ee6d2c372d8f403a3c3f3109a13d1/pangeo_forge_recipes/recipes/reference_hdf_zarr.py#L25) recipe class today, and it seemed that this change was needed to concatenate a list of files with varying lengths in the time dimension.

Thoughts, @martindurant ?

cc @rabernat

**Comment by martindurant** [comment]
True, every example so far has been regular in this sense. I am still missing a fully general way to form the correct zarr key names for a set of input nd-arrays; given that the concat coord(s) may be coordinates of those arrays or not, and you might have multiple variables too. The code in the combine module basically deals with the set of examples that I've seen so far.
